### PR TITLE
Fixes #3: Transifex Integration

### LINKED ISSUE #3: translate beta7
would be great if we could translate [beta7](https://github.com/flarum/flarum-ext-english/releases/tag/v0.1.0-beta.7) and release a version.

@Alir3z4 is this possible with transifex?

### PR DESCRIPTION
**Pull request by Alir3z4**
With some more translations.
The changes in folder structure are required to make it completely compatible with Transifex or mostly any other translations management tools.

Fixes #3

### CODE REVIEW

**Comment by Alir3z4** [comment]
With this getting merged, we can easily fly to moon and start making much easier progress with translating

File: src/Listener/AddLanguagePack.php [hunk 1]
```diff
@@ -14,6 +14,6 @@ public function subscribe(Dispatcher $events)
 
     public function addLanguage(ConfigureLocales $event)
     {
-        $event->loadLanguagePackFrom(__DIR__.'/../..');
+        $event->loadLanguagePackFrom(__DIR__.'/../../fa_IR/');
```
Comment (Alir3z4) [inline src/Listener/AddLanguagePack.php:17]: I'm not a PHP hombre or even a Flarum dev dude. Is this correct ?

**Comment by sijad** [comment]
can't `source_file` pints to another repository or url?
I'm not against keeping sources in same repository (I also think it's good practice), but I guess it's better to keep structure same as Flarum?

and some comments has been removed from translations, we need to revert them back, I want to help, please let me know how do you do that.

**Comment by Alir3z4** [comment]
> can't source_file pints to another repository or url?

We can use git submodules to load the whole `https://github.com/flarum/flarum-ext-english` repo under that sources directly. However this will add extra complication to the whole flow an how to maintain it each time we're working with it.

As you mentioned, keeping the source in the same repo is a good practice and I'm not sure why Flarum has not followed the same workflow. The way flarum is actually organized in the localization is making it confusing to get localization and i18n done properly and I see they haven't moved the needle much since the beginning and still their documentation regarding the translation is blank.

Regarding the comments yes, I see they have been removed. However when working with translation, only the Source file needs to have the context and comments or any related idea for translators. In our scenario, en_US folder has all the comments and necessary data for someone starting to translate, but any other folder that is not the source, such as fa_IR is simply output and the final translation. The final translation however shouldn't contain comments from the source but comments on its own. The format is YML and I'm not sure how much translation tools are compatible with keeping the comments of such format, because in PO format the annotation is straightforward and it widely used, therefore translation tools work with it. 

I have contacted the Transifex support regarding keeping the comments in the translation files from the source and I'll wait to see what their comment is.


Anyway, I understand these changes that I made might be too much and bring complexity to the project, but in the long run they're a must whenever doing localization.

Please let me know how would you like me to make changes so you can merge it.

**Comment by sijad** [comment]
keeping source files seems right to me I guess, let do that.

for commnets (https://docs.transifex.com/formats/yaml):

> For use: Get a file that only contains translated entries. Comments will not be included in the downloaded file unless you download the source language file for use. If you want comments to be included in a translation file, download the version for translation instead.

is it possible to push `version for translation` in the repo instead?

**Comment by Alir3z4** [comment]
Good findings.
Let me dig with the tx to see how it's done, otherwise we have to do it
manually.

I'll update you shortly

**Comment by sijad** [comment]
we probably wont pull changes from Transifex more than 2-3 times per version which can be done manually, I'm not sure if it does worth to changing the structure and losing comments.

**Comment by Alir3z4** [comment]
When translating, the comments should be in the source, but they're not necessary needed in the translated file.
However, I tried to get them by tx client, but that didn't work automatically so I downloaded all of them manually and put them into the repo.

Please have a look and let me know if you need any other changes applied.
Also, please consider this https://github.com/sijad/flarum-ext-persian/pull/5#discussion_r181559867

**Comment by sijad** [comment]
I guess we should revert back to old structure and pull changes from Transifex manually, this way we also can use same Transifex project for other Flarum extensions, please let me know what do you think?

**Comment by sijad** [comment]
ping

**Comment by sijad** [comment]
hey, I've just tested your changes, got this error:

```
Uncaught RuntimeException: Language pack fa_IR must define "extra.flarum-locale.code" in composer.json.
```

it seems Flarum can't work with this structure (`fa_IR/locale`), there might be a workaround for this but I still think it's better to keep the things like original.

everything else seems good, would you please revert back original structure?

**Comment by Alir3z4** [comment]
Hey @sijad, I've been swamped recently.
I will get back to this soon and make the changes.

**Comment by sijad** [comment]
@Alir3z4 please checkout #6

**Pull request closed without merging**